### ISSUE TITLE
[Feature Request] Layers in canvas

### ISSUE THREAD

**Issue by santiagobasulto** [excalidraw/excalidraw#6266]
I'd like to have a way to draw in different layers. That way, I can hide/display a layer and its content and create diagrams that are "stacked". For example, in my following drawing, I'd make all the blue annotations in a different layer, so I can toggle it on and off while I'm presenting.

<img width="1390" alt="image" src="https://user-images.githubusercontent.com/872296/220242180-e5dca01f-1937-4aeb-918e-f7767ced9ed2.png">

I could also "automate" the layers? By saying:

* Start with Layer 1. Stop for 3 seconds
* Display layer 2. Move to X, Y. Stop for 2 seconds
* Display layers 3 and 4. Zoom to X, Y. Stop for 4 seconds

**Comment by 7flash** [comment]
Yes! And when it gets really interesting that's when layouts can be automatically switched as you zooming into content to reveal another diagram explaining the same concept in more details, - i.e. Stretch-Text. There might be an issue regarding performance, but it can be resolved by actually deleting elements while persisting them in the database, as it currently works in my prototype of holonic mindmaps based on excalidraw, - and I can try to prepare a PR if it makes sense.

**Comment by chaecramb** [comment]
Really interesting idea @7flash, would love to see something like that. 

As a user of @zsviczian's Obsidian Excalidraw I was just thinking about a use case for layers for learning. By being able to embed an Excalidraw drawing in a note with different layers in place you basically have an excellent system for visual flashcards. 

For example you can have a layer that occludes part of the image as a question, and then the base layer which is unoccluded as the answer. Implementing this with layers means many different 'views' would be possible on a single drawing.

**Comment by 7flash** [comment]
This is my progress in implementation of multi-window layers

Uploading feb20-first.mp4…

**Comment by 7flash** [comment]
This is my progress in implementation of multi-window layers


https://github.com/excalidraw/excalidraw/assets/4569866/a561f785-0d64-4cc8-a21d-7655fab3604d

**Comment by mairav** [comment]
I would also love the option to name layers and hide or show them. I am drawing construction plans and I need to create different stages of the work, or different materials, in layers.

The zooming option would awesome too!

**Comment by yachal-upson** [comment]
+1
Wanted to bring some work over from Inkscape (a landscape design), to hand sketch some markup on my Samsung Tab. 
Then found myself quickly limited by lack of layers.

LOVE what excalidraw has brought the community. Thank you!
But layers. Fundamental.
